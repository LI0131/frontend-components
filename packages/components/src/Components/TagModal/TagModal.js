import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './tagModal.scss';
import { Modal } from '@patternfly/react-core';
import {
    Table,
    TableHeader,
    TableBody,
    sortable,
    SortByDirection,
    headerCol,
    TableVariant,
    expandable,
    cellWidth,
    textCenter,
  } from '@patternfly/react-table';
  

export default class TagModal extends React.Component {
    constructor(props) {
        super(props);
        let rows = []
        //this bit should be replaced with an API call when the backend is ready
        if(props.systemName == 'paul.localhost.com') {
            rows = [
                {
                    cells: ['environment=production', 'AWS']
                },
                {
                    cells: ['owner=finanace', 'AWS']
                },
                {
                    cells: ['pants=worn', 'AWS']
                },
                {
                    cells: ['fit=F R E S H', 'AWS']
                },
                {
                    cells: ['clams=casino', 'AWS']
                },
                {
                    cells: ['OS=RHEL', 'AWS']
                },
                {
                    cells: ['computer=fast', 'AWS']
                },
                {
                    cells: ['Iam=SPEED!', 'AWS']
                },
                {
                    cells: ['beans=food', 'AWS']
                },
                {
                    cells: ['thing=otherthing', 'AWS']
                },
                {
                    cells: ['cheese=cheddar', 'AWS']
                }
            ];
        } else {
            rows = [
                {
                    cells: ['environment=staging', 'AWS']
                },
                {
                    cells: ['owner=engineering', 'AWS']
                }
            ];
        }

        this.state = {
            columns: [
                { title: 'Name' },
                { title: 'Tag Source' }
            ],
            rows: rows
        };
    }
    render() {
        return(
            <Modal
            width={'50%'}
            title={"Tags for " + this.props.systemName}
            isOpen={this.props.modalOpen}
            onClose={this.handleModalToggle}
            isFooterLeftAligned
            >
                <Table cells={this.state.columns} rows={this.state.rows}>
                    <TableHeader />
                    <TableBody />
                </Table>
            </Modal>
        )
    }
}