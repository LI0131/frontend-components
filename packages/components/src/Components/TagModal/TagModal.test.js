import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TagModal from './TagModal';

describe('TagCount component', () => {
    it('Render the modal open with 11 tags', () => {
        const wrapper = shallow(<TagModal modalOpen={true} systemName={'paul.localhost.com'} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
