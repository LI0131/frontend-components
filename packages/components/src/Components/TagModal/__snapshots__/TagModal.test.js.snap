// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCount component Render the modal open with row of tags 1`] = `
<Modal
  actions={Array []}
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isFooterLeftAligned={true}
  isLarge={false}
  isOpen={true}
  isSmall={false}
  onClose={[Function]}
  rows={
    Array [
      Array [
        "key",
        "value",
      ],
      Array [
        "thing",
        "otherthing",
      ],
    ]
  }
  title="Tags for paul.localhost.com"
>
  <Table
    borders={true}
    cells={
      Array [
        Object {
          "title": "Name",
        },
        Object {
          "title": "Tag Source",
        },
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Array [
          "key",
          "value",
        ],
        Array [
          "thing",
          "otherthing",
        ],
      ]
    }
    variant="compact"
  >
    <TableHeader
      className=""
    />
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</Modal>
`;

exports[`TagCount component Render the modal with a child component 1`] = `
<Modal
  actions={Array []}
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isFooterLeftAligned={true}
  isLarge={false}
  isOpen={true}
  isSmall={false}
  onClose={[Function]}
  title="Tags for paul.localhost.com"
>
  <h1>
    I am a chile component
  </h1>
  <Table
    borders={true}
    cells={
      Array [
        Object {
          "title": "Name",
        },
        Object {
          "title": "Tag Source",
        },
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    variant="compact"
  >
    <TableHeader
      className=""
    />
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</Modal>
`;
